src/meteodatalab/operators/relhum.py

```diff
@@ -43,18 +43,26 @@ def relhum(
     max = 100 if clipping else None
 
     phase_conditions = {
-        "water": {"func": pv_sw(t), "shortName": "RELHUM"},
-        "ice": {"func": pv_si(t), "shortName": "RH_ICE"},
-        "water+ice": {"func": pv_sm(t), "shortName": "RH_MIX_EC"},
+        "water": {"shortName": "RELHUM"},
+        "ice": {"shortName": "RH_ICE"},
+        "water+ice": {"shortName": "RH_MIX_EC"},
     }
 
     if phase not in phase_conditions:
         raise ValueError("Invalid phase. Phase must be 'water', 'ice', or 'water+ice'.")
 
-    result = (100.0 * qv / qv_pvp(phase_conditions[phase]["func"], p)).clip(0, max)
+    q = r / (1 + r)
+
+    pb, tb, qb = xr.broadcast(p, t, q)
+
+    phase_for_svp = "mixed" if phase == "water+ice" else phase
```

Comment:
I think this can be added to the `phase_conditions` mapping to make it more explicit